README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
+```
+
+GET document
+
+```JavaScript
+const res = await fetch('earthstar://+gardening.xxxxx/example')
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Get the author for this document
+res.headers.get('earthstar-author') === '@bbbb.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+
+// Get the content of the document back out.
+// If it's JSON, you should use `.json()`
+const content = await res.text()
```

Comment:
When `GET`ting a specific document we can use the file extension to guess the MIMEType.

For now if it's a binary mimetype like JPG, the data will have been stored base64-encoded, so the browser will need to un-encode it for display.

Reply:
Might be good to brainstorm the set of mimetypes which are binary or text. 🤔

Alternately, we could leave the base64 decode to be opt-in via the URL? Like `earthstar://+example.example/path/to/image.jpg?base64`

Reply:
How about `?encoding=base64` or `?encoding=utf8` so it can be combined with the other query options?

Default is `utf8` unless it's on a short list of common filetypes, probably we could list the 10 or 20 most common image / video / audio formats and that would be good enough

https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#common_image_file_types

https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Containers#browser_compatibility